fluent-syntax/src/parser.js

```diff
@@ -411,65 +409,170 @@ export default class FluentParser {
     return new AST.NumberLiteral(num);
   }
 
-  getValue(ps) {
-    if (ps.currentChar === "{") {
+  // getValue distinguished between patterns which start on the same line as the
+  // identifier (a.k.a. inline signleline patterns and inline multiline
+  // patterns) and patterns which start on a new line (a.k.a. block multiline
+  // patterns). The distinction is important for the dedentation logic: the
+  // indent of the first line of a block pattern must be taken into account when
+  // calculating the maximum common indent.
+  getValue(ps, {allowVariantList}) {
+    ps.peekBlankInline();
+    if (ps.isValueStart()) {
+      ps.skipToPeek();
+      return this.getValueType(
+        ps, {isBlock: false, allowVariantList});
+    }
+
+    ps.peekBlankBlock();
+    if (ps.isValueContinuation()) {
+      ps.skipToPeek();
+      return this.getValueType(ps, {isBlock: true, allowVariantList});
+    }
+
+    return null;
+  }
+
+  // Parse a VariantList (if allowed) or a Pattern.
+  getValueType(ps, {isBlock, allowVariantList}) {
+    ps.peekBlankInline();
+    if (allowVariantList && ps.currentPeek === "{") {
+      const start = ps.peekOffset;
       ps.peek();
       ps.peekBlankInline();
-      if (ps.isNextLineVariantStart({skip: false})) {
-        return this.getVariantList(ps);
+      if (ps.currentPeek === EOL) {
```

Comment:
how does this handle `\r\n`?

Reply:
`ParserStream` normalizes `CRLF` to `LF`:

https://github.com/projectfluent/fluent.js/blob/e5cf0c0bbe69d7e97812f1241f18caef103a653c/fluent-syntax/src/stream.js#L13-L24